src/Bridge/Laravel/Http/Middlewares/AuditMiddleware.php

```diff
@@ -0,0 +1,86 @@
+<?php
+declare(strict_types=1);
+
+namespace LoyaltyCorp\Auditing\Bridge\Laravel\Http\Middlewares;
+
+use Closure;
+use Illuminate\Http\Request;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Symfony\Bridge\PsrHttpMessage\HttpMessageFactoryInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+class AuditMiddleware
+{
+    /**
+     * @var \Symfony\Bridge\PsrHttpMessage\HttpMessageFactoryInterface
+     */
+    private $psr7Factory;
+
+    /**
+     * AuditMiddleware constructor.
+     *
+     * @param \Symfony\Bridge\PsrHttpMessage\HttpMessageFactoryInterface $psr7Factory
+     */
+    public function __construct(HttpMessageFactoryInterface $psr7Factory)
+    {
+        $this->psr7Factory = $psr7Factory;
+    }
+
+    /**
+     * Handle the request and response and convert them to psr7 to be logged
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param \Closure $next
+     *
+     * @return mixed
+     */
+    public function handle(Request $request, Closure $next)
+    {
+        try {
+            $response = $next($request);
+        } catch (\Exception $exception) {
+            // TODO: returning null until a decision is made on how to handle exceptions
+            $response = null;
+        }
+
+        $psrRequest = $this->createPsr7Request($request);
+        $psrResponse = null;
+        if (($response instanceof Response) === true) {
+            $psrResponse = $this->createPsr7Response($response);
+        }
+
+        // if response is already an instance of psr7 response
+        if (($response instanceof ResponseInterface) === true) {
+            $psrResponse = $response;
+        }
+
+        // TODO: space to plug in service (PYMT-713) and use $psrRequest and $psrResponse
```

Comment:
When plugging in PYMT-713, we're going to need to wrap it in a try/catch and report any exceptions that were generated then continue and return the original response.